view/frontend/web/js/view/payment/method-renderer/coingate-method.js

```diff
@@ -1,119 +1,69 @@
-/**
- * CoinGate JS
- *
- * @category    CoinGate
- * @package     CoinGate_Merchant
- * @author      CoinGate
- * @copyright   CoinGate (https://coingate.com)
- * @license     https://github.com/coingate/magento2-plugin/blob/master/LICENSE The MIT License (MIT)
- */
- /*browser:true*/
- /*global define*/
- define(
- [
-     'jquery',
-     'Magento_Checkout/js/view/payment/default',
-     'Magento_Checkout/js/action/place-order',
-     'Magento_Checkout/js/action/select-payment-method',
-     'Magento_Customer/js/model/customer',
-     'Magento_Checkout/js/checkout-data',
-     'Magento_Checkout/js/model/payment/additional-validators',
-     'mage/url',
- ],
- function (
-     $,
-     Component,
-     placeOrderAction,
-     selectPaymentMethodAction,
-     customer,
-     checkoutData,
-     additionalValidators,
-     url) {
-     'use strict';
-
-
-     return Component.extend({
-         defaults: {
-             template: 'CoinGate_Merchant/payment/coingate-form'
-         },
-
-         placeOrder: function (data, event) {
-
-             var test = $.ajax({
-
-                 url: url.build('coingate/payment/testConnection'),
-                 type: 'POST',
-                 async: false,
-                 dataType: 'json'
-             });
-
-            var result = null;
-
-             test.done(function (response)  {
-
-                 if (response.status === false) {
-
-                     alert(response.reason + "\n Please contract merchant");
-                     location.reload();
-                     result = false;
-                 }
-             });
-
-            if (result === false){
-
+define([
```

Comment:
will all this javascript be compatible with various themes and one-click checkout plugins that many users use ?

Reply:
yes, we have tested with two popular vendors - amasty and mageplaza